app/lib/memory_repository.rb

```diff
@@ -177,6 +177,7 @@ def get_revision_by_timestamp(at_or_earlier_than, path = nil, later_than = nil)
     end
 
     (@revision_history + [@current_revision]).reverse_each do |revision|
+      return revision if revision.server_timestamp.nil?
```

Comment:
Can you explain why we need to return early here?

Reply:
Correct me if i am wrong
In the next 2 lines we have to compare **revision.server_timestamp** value with other values. If *revision.server_timestamp* value is nil it will raise error.